internal/clientio/resp.go

```diff
@@ -152,17 +152,42 @@ func encodeString(v string) []byte {
 	return []byte(fmt.Sprintf("$%d\r\n%s\r\n", len(v), v))
 }
 
+// encodeBool encodes bool as simple strings
+func encodeBool(v bool) []byte {
+	var val string
+	switch v {
+	case true:
+		val = "true"
+	case false:
+		val = "false"
+	}
+	return []byte(fmt.Sprintf("+%v\r\n", val))
+}
+
 func Encode(value interface{}, isSimple bool) []byte {
 	switch v := value.(type) {
 	case string:
-		if isSimple {
+		// encode as simple strings
+		if isSimple || v == "[" || v == "{" {
```

Comment:
`v` will store the name of the type (string, int, int8, etc.). How can it be `[` or `{`?